scripts/build_locally.py

```diff
@@ -14,225 +14,205 @@
         "--verbose",
-        help="Build using vebose makefile mode",
         dest="verbose",
         action="store_true",
+        help="Enable verbose makefile output",
+    )
+
+    p.add_argument(
+        "--no-level-zero",
+        dest="no_level_zero",
+        action="store_true",
+        default=False,
+        help="Disable Level Zero backend (deprecated: use --target-level-zero "
+        "OFF)",
     )
-    driver.add_argument(
+    p.add_argument(
+        "--target-level-zero",
+        action="store_true",
+        help="Enable Level Zero backend explicitly",
+    )
+
+    p.add_argument(
         "--cmake-opts",
-        help="Options to pass through to cmake",
-        dest="cmake_opts",
-        default="",
         type=str,
+        default="",
+        help="Additional options to pass directly to CMake",
     )
-    driver.add_argument(
+
+    p.add_argument(
         "--target-cuda",
         nargs="?",
         const="ON",
-        help="Enable CUDA target for build; "
-        "optionally specify architecture (e.g., --target-cuda=sm_80)",
         default=None,
-        type=str,
+        help="Enable CUDA build. Architecture is optional to specify.",
     )
-    driver.add_argument(
+    p.add_argument(
         "--target-hip",
         required=False,
-        help="Enable HIP target for build. "
-        "Must specify HIP architecture (e.g., --target-hip=gfx90a)",
         type=str,
+        help="Enable HIP backend. Architecture required to be specified.",
+    )
+
+    p.add_argument(
+        "--clean",
+        action="store_true",
+        help="Remove build dir before rebuild",
+    )
+    p.add_argument(
+        "--skip-editable",
+        action="store_true",
+        help="Skip pip editable install step",
+    )
+
+    return p.parse_args()
+
+
+def main():
+    if sys.platform not in ["cygwin", "win32", "linux"]:
+        err(f"{sys.platform} not supported", "build_locally")
+    args = parse_args()
+    setup_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+    c_compiler, cxx_compiler = resolve_compilers(
+        args.oneapi, args.c_compiler, args.cxx_compiler, args.compiler_root
     )
-    args = parser.parse_args()
-
-    args_to_validate = [
-        "c_compiler",
-        "cxx_compiler",
-        "compiler_root",
-    ]
-
-    if args.oneapi or (
-        args.c_compiler is None
-        and args.cxx_compiler is None
-        and args.compiler_root is None
-    ):
-        args.c_compiler = "icx"
-        args.cxx_compiler = "icpx" if "linux" in sys.platform else "icx"
-        args.compiler_root = None
+
+    # clean build dir if --clean set
+    if args.clean:
+        clean_build_dir(setup_dir)
+
+    if args.no_level_zero and args.target_level_zero:
+        err(
+            "Cannot combine --no-level-zero and --target-level-zero",
+            "build_locally",
+        )
+
+    # Level Zero state (on unless explicitly disabled)
+    if args.no_level_zero:
```

Comment:
I wonder, if `--no-level-zero` is deprecated, should we base the logic mostly on `--target-level-zero` option then?
Should we consider `level_zero_enabled = False` if `--target-level-zero` is not passed?